examples/tp/tm_high_order.py

```diff
@@ -0,0 +1,527 @@
+plt.ylim([-0.1,1.1])
+plt.xlim([320,520])
+plt.plot(x, y)
+plt.xlabel("Timestep")
+plt.ylabel("Prediction Accuracy")
+plt.title("Fig. 5: Accuracy drops even more when more noise is added to input")
+plt.savefig("figure_5")
+plt.close()
+
+# Let's present the original sequence to the TM in order to re-learn it. After this, the
+# TM will predict accurately the sequence again, and its predictions will not include 'noisy'
+# columns anymore.
+for t in range(25):
+  tm.reset()
+  for k in range(4):
+    tm.compute(set(seq2[k][:].nonzero()[0].tolist()), learn=True)
+    activeColumnsIndeces = [tm.columnForCell(i) for i in tm.getActiveCells()]
+    predictedColumnIndeces = [tm.columnForCell(i) for i in tm.getPredictiveCells()]
+    actColState = ['1' if i in activeColumnsIndeces else '0' for i in range(tm.numberOfColumns())]
+    actColStr = ("".join(actColState))
+    currCols = [1 if i in activeColumnsIndeces else 0 for i in range(tm.numberOfColumns())]
+    acc = accuracy(currCols, predCols)
+    x.append(ts)
+    y.append(acc)
+    ts += 1
+    predCols = [1 if i in predictedColumnIndeces else 0 for i in range(tm.numberOfColumns())]
+    predColState = ['1' if i in predictedColumnIndeces else '0' for i in range(tm.numberOfColumns())]
+    predColStr = ("".join(predColState))
+
+# We will observe how the prediction accuracy gets back to 1.0 (not in-between sequences)
+# as the TM re-learns the original sequence.
+plt.ylim([-0.1,1.1])
+plt.xlim([520,620])
+plt.plot(x, y)
+plt.xlabel("Timestep")
+plt.ylabel("Prediction Accuracy")
+plt.title("Fig. 6: When noise is suspended, accuracy is restored")
+plt.savefig("figure_6")
+plt.close()
+
+# The TM restores its prediction accuracy and it can be seen when presented with the individual characters.
+# There's no noisy columns being predicted.
+print ""
+print "-"*50
+print "After presenting noisy input to the TM, we present the original sequence in"
+print "order to make it re-learn XBCY. We verify that this was achieved by presenting"
+print "the TM with the individual characters and observing its output. Again, we can"
+print "see that the 'noisy' columns are not being predicted anymore, and that the"
+print "prediction accuracy goes back to 1.0 when the sequence is presented (Fig 6)."
+print "-"*50
+print ""
+
+for k in range(6):
+  print "--- " + "ABCDXY"[k] + " ---"
+  tm.compute(set(seqT[k][:].nonzero()[0].tolist()), learn=False)
+  activeColumnsIndeces = [tm.columnForCell(i) for i in tm.getActiveCells()]
+  predictedColumnIndeces = [tm.columnForCell(i) for i in tm.getPredictiveCells()]  
+  currCols = [1 if i in activeColumnsIndeces else 0 for i in range(tm.numberOfColumns())]
+  predCols = [1 if i in predictedColumnIndeces else 0 for i in range(tm.numberOfColumns())]
+  print("Active cols: " + str(np.nonzero(currCols)[0]))
+  print("Predicted cols: " + str(np.nonzero(predCols)[0]))
+  print ""
+
+# PART 4. Now, we will present both sequences ABCD and XBCY randomly to the TM.
```

Comment:
This is a bit confusing. Are you starting from a fresh TM or learning on top of the previous TM?  In these examples you are no longer sending a reset in between right? You might want to make all this clear.